platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdmin.java

```diff
@@ -427,18 +433,32 @@ public void updateForLocation(String pid, String location,
             throw new IOException("Argument configurationTable cannot be null");
         }
 
-        // sanity check to make sure no values are
-        // null
-        for (Entry<String, Object> curEntry : configurationTable.entrySet()) {
-            Object value = curEntry.getValue();
-            if (value == null || (value instanceof ArrayList && CollectionUtils.sizeIsEmpty(value))) {
-                curEntry.setValue("");
+        // find the config's metatype so we can check cardinality
+        Configuration config = configurationAdmin.getConfiguration(pid, location);
+        List<Map<String, Object>> services = listServices();
+        List<Map<String, Object>> tempMetatype = null;
+        for (Map<String, Object> service : services) {
+            String id = (String) service.get(ConfigurationAdminExt.MAP_ENTRY_ID);
+            if (id.equals(config.getPid()) || (id.equals(config.getFactoryPid())
+                    && (boolean) service.get(ConfigurationAdminExt.MAP_FACTORY))) {
```

Comment:
Boolean.valueOf(String.valueOf(thing)) might be safer here than a cast to avoid an instanceof check.